### PR TITLE
Use messybrainz lib

### PR DESCRIPTION
**Pull request by mayhem**
Avoid making massive numbers of calls to messybrainz by using a local library.

### CODE REVIEW

File: webserver/views/api.py [hunk 1]
```diff
@@ -52,59 +54,21 @@ def submit_listen():
...
-                listen['track_metadata']['additional_info']['artist_msid'] = messybrainz_resp['artist_msid']
-            except KeyError:
-                current_app.logger.error("MessyBrainz did not return a proper set of ids")
-                raise InternalServerError
-
-            try:
-                listen['track_metadata']['additional_info']['release_msid'] = messybrainz_resp['release_msid']
-            except KeyError:
-                pass
+    if len(msb_listens):
```
Comment (gentlecat) [inline webserver/views/api.py]: I think you can just do `if msb_listens:` there.

File: webserver/views/api.py [hunk 2]
```diff
@@ -156,54 +120,97 @@ def _validate_auth_header():
...
+                messy_dict['release_mbid'] = ai['release_mbid'] 
+            if 'recording_mbid' in ai:
+                messy_dict['recording_mbid'] = ai['recording_mbid'] 
+            if 'track_number' in ai:
+                messy_dict['track_number'] = ai['track_number'] 
+            if 'spotify_id' in ai:
+                messy_dict['spotify_id'] = ai['spotify_id'] 
+        msb_listens.append(messy_dict)
+
+    print msb_listens
```
Comment (gentlecat) [inline webserver/views/api.py]: `print(msb_listens)` (py3)

Comment (gentlecat) [inline webserver/views/api.py]: Ignore me.

File: webserver/views/api.py [hunk 3]
```diff
@@ -156,54 +120,95 @@ def _validate_auth_header():
...
+                current_app.logger.error("Kafka listens write error: " + str(sys.exc_info()[0]))
+                raise InternalServerError("Cannot record listen at this time.")
+
+
+def _messybrainz_lookup(listens):
+
+    msb_listens = []
+    for listen in listens:
+        messy_dict = {
+            'artist': listen['track_metadata']['artist_name'],
```
Comment (alastair) [inline webserver/views/api.py]: You never check that `listen['track_metadata']` is present in validation

Comment (alastair) [inline webserver/views/api.py]: yes you do. sorry

**Pull request merged**